### PR TITLE
Add prow node e2e test.

### PR DESCRIPTION
**Pull request by Random-Liu**
Add node e2e test framework for prow.
@krzyzacy @yguo0905

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

**Comment by Random-Liu** [comment]
@krzyzacy @yguo0905 Could you help me review this?

**Comment by krzyzacy** [comment]
assume you'll have a `Dockerfile` for runner.sh?

File: test/e2e_node/configure.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+tar xvf "${TARBALL}"
+
+# EXTRA_INIT_SCRIPT is the name of the extra init script after being downloaded.
+EXTRA_INIT_SCRIPT="extra-init.sh"
+# EXTRA_INIT_SCRIPTINIT_SCRIPT_METADATA is the metadata key of init script.
+EXTRA_INIT_SCRIPT_METADATA="extra-init-sh"
+
+# Check whether extra-init-sh is set.
+if ! curl --fail --retry 5 --retry-delay 3 --silent --show-error -H "X-Google-Metadata-Request: True" "${ATTRIBUTES}/" | \
+  grep -q "${INIT_SCRIPT_METADATA}"; then
```
[the remark this answers is not in the document]
Comment (Random-Liu) [reply test/e2e_node/configure.sh]: Done.

**Comment by Random-Liu** [comment]
> assume you'll have a Dockerfile for runner.sh?

I get rid of the runner, we should be able to use the original runner directly.

**Comment by Random-Liu** [comment]
@mikebrow Thanks! I'll merge this PR. To really know whether a test works or not in Kubernetes test infra, the only way is to merge and try it. :p

**Pull request merged**